tee-worker/omni-executor/intent/executors/cross-chain/src/lib.rs

```diff
@@ -504,27 +579,386 @@ impl<
+						};
+						self.pumpx_api.cross_fail(&access_token, data).await.map_err(|_| {
+							log::error!("Failed to notify pumpx-signer");
+						})?;
+						// TODO: Figure out how to transfer back the asset to the omni account
+						// check https://developers.binance.com/docs/wallet/capital/withdraw
+
+						return Err(());
+					};
+
+					let mut trade_success = false;
+					let mut to_amount = 0;
+					loop {
+						let trade_order = self
+							.binance_api
+							.spot_trading()
+							.get_order(&trade_symbol, Some(binance_order.order_id), None, None)
+							.await
+							.map_err(|_| {
+								log::error!("Failed to get binance order");
+							})?;
+
+						match trade_order.status {
+							BinanceOrderStatus::FILLED => {
+								log::info!("Binance order filled");
+								for fill in trade_order.fills.unwrap() {
+									let fill_price: u128 = fill.price.parse().unwrap();
+									let fill_amount: u128 = fill.qty.parse().unwrap();
+									to_amount += fill_price * fill_amount;
+								}
+								trade_success = true;
+								break;
+							},
+							BinanceOrderStatus::CANCELED => {
+								log::error!("Binance order canceled");
+							},
+							BinanceOrderStatus::REJECTED => {
+								log::error!("Binance order rejected");
+							},
+							BinanceOrderStatus::EXPIRED => {
+								log::error!("Binance order expired");
+							},
+							BinanceOrderStatus::EXPIRED_IN_MATCH => {
+								log::error!("Binance order expired in matching");
+							},
+							_ => {
+								log::debug!("Binance order status: {:?}", trade_order.status);
+							},
+						}
+						//todo: how long we wait ?
+						sleep(Duration::from_millis(500)).await;
+					}
+					if !trade_success {
+						log::error!("Binance order failed");
+						let data = CrossOrderFailData {
+							request_id: intent_id,
+							// TODO: is this a user facing error? what should we return?
+							fail_reason: "Binance order failed".to_string(),
+						};
+						self.pumpx_api.cross_fail(&access_token, data).await.map_err(|_| {
+							log::error!("Failed to notify pumpx-signer");
+						})?;
+						// TODO: Figure out how to transfer back the asset to the omni account
+						// check https://developers.binance.com/docs/wallet/capital/withdraw
+
+						return Err(());
+					}
+
+					debug!("Total traded on binance: {:?}", to_amount);
+
+					debug!("Getting {:?} nonce for payout request", payout_address);
+					// 4. Call accounting contract on BSC
+					let user_nonce =
+						self.accounting_contract_client.get_nonce(payout_address).await.map_err(
+							|_| {
+								log::error!("Failed to get nonce");
+							},
+						)?;
+
+					debug!("Received {:?} nonce", user_nonce);
+
+					debug!("Calling accounting contract payout with address {:?}, nonce {:?} and amount {:?}", payout_address, user_nonce, to_amount);
+					//todo : make sure we payout correct amount
+					self.accounting_contract_client
+						.execute_pay_out_request(
```

Comment:
Second, is the tx actually signed? 😂

Reply:
It will faill, we need to provide wallet here :

```
			let ethereum_rpc_provider = ethereum_rpc::AlloyRpcProvider::new(&args.ethereum_url);
			let accounting_contract_client = AccountingContractClient::new_with_wallet(
				ethereum_rpc_provider,
				"TODO:get contract address".parse().unwrap(),
			);
```